### ISSUE TITLE
MAINT: investigate potential file descriptor leaks

### ISSUE THREAD

**Issue by lucascolley** [scipy/scipy#19845]
Following gh-19842 and gh-19844, it was reported in gh-19844 for there are still some potential file descriptor leaks found with the following hack:

```
from test.support.os_helper import fd_count

@pytest.fixture(autouse=True)
def fd_leak():
    before = fd_count()
    yield
    assert fd_count() == before
```

It would be good to see whether any of these can be patched up, as it can cause issues when garbage collection is less aggressive.

**Comment by efrat99** [comment]
I’m currently working on a fix for this issue and started a systematic investigation in scipy.stats, focusing on resource cleanup and potential file descriptor leaks.

So far, introducing more explicit cleanup patterns has already led to a noticeable reduction in leaked handles during local testing.

Before going further, @lucascolley, I wanted to check whether this issue is still considered a priority, and whether there are particular areas or edge cases you’d recommend paying special attention to.

I’m planning to submit a PR once the investigation is complete.

**Comment by lucascolley** [comment]
I would say it is low priority but fixes would still be welcome. cc @nickodell who may also be interested in this kind of bug

**Comment by nickodell** [comment]
BTW, taking a brief look at this, there are three broad categories of leaks here.

 * MessageStream temporary files. This is for external libraries that write errors to a `FILE*` (unuran, qhull) on platforms without support for open_memstream.
 * Multiprocessing pools that don't get cleaned up. This is usually a problem with the test case hanging onto a reference to the pool. Haven't looked at each case to be sure. The subprocesses have a two file descriptors attached to them for the purpose of sending work / receiving results.
 * Multiprocessing's resource tracker. The first test to make use of a cross process lock or cross process queue will create a resource tracker. The resource tracker's purpose is to clean up any remaining locks after every subprocess exits. It is intended to live for the entire life of your process. This means that some tests will increase the FD count if they are the first test to use the resource tracker. Options are to ignore it, start the resource tracker before testing, and to kill the resource tracker before each test exits.
